### PR TITLE
docs(bodiless-documentation): Tailwind/Custom Css guide for Site builder

### PR DESCRIPTION
**Pull request by hvanyo**
## Changes
Documentation for Tailwind
Custom Css
Responsiveness

### CODE REVIEW

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Responsiveness
+
+BodilessJS is in itself is not very opinionated about the responsiveness but there
+are some components and features that do provide specific responsive behaviors.
+
+BodilessJS primarily uses [TailwindCSS](https://tailwindcss.com) to give the
+site its reponsiveness. While not a requirement to use, its included within the
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:7]:
```suggestion
site its reponsiveness. While not a requirement to use, it's included within the
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+Example
+[Test Site](https://github.com/johnsonandjohnson/Bodiless-JS/tree/master/examples/test-site)
+and
+[Starter Site](https://github.com/johnsonandjohnson/Bodiless-JS/tree/master/examples/starter).
+
+### Site Editing
+
+The BodilessJS site editing interface was designed to be edited in desktop
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:15]:
```suggestion
The BodilessJS site editing interface was designed for desktop
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+screen for productivity.
+
+!> **Important** The BodilessJS editor and its forms are designed to work
+on a screensize of a minimum of 1024px. This is not to say it can't be used on a
+smaller device, but some editor forms may appear off the device screen causing
+extra scrolling.
+
+## Breakpoints
+By default, the breakpoints are set by Tailwind and are defined by TailwindCSS
+default's
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:26]:
```suggestion
default's:
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+| Small     | Medium    | Large      | Extra Large | 
+| --------- | --------- | ---------- | ----------- |
+| <= 640px  | 641-768px | 769-1024px | 1280px      |
+
+A site builder can changes by making the modifications in the sites
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:32]:
```suggestion
A site builder can change the breakpoints by modifying the site's
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+`tailwind.config.js` by following the directions specified in
+[Tailwind Breakpoints](https://tailwindcss.com/docs/breakpoints/).
+
+?> **Tip** If a site developer does modify and/or add extra breakpoints, please
+review the next section to determine if there needs to be update to any
+component's behavior.
+
+## Components with Custom a Responsive Behavior
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:40]:
```suggestion
## Components with Custom Responsive Behavior
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+At times, there is need to define a specific responsive behavior that can't be
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:42]:
```suggestion
At times, you may need to define a specific responsive behavior that can't be
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+defined by tailwind classes by itself and should be included within the
+component. BodilessJS provides a mechanism to allow the site to handle the
+custom responsive behaviors.
+
+Within the Starter site, the
+[`src/components/Page/index.tsx`](https://github.com/johnsonandjohnson/Bodiless-JS/tree/master/examples/starter/src/components/Page/index.tsx)
+will read in the Tailwind breakpoints via
+[resolveConfig()](https://tailwindcss.com/docs/configuration/#referencing-in-javascript)
+set in the Tailwind.config.js (see previous section). These breakpoints are
+exported as `breakpoints` object that contain the site's configuration.
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:52]:
```suggestion
exported as `breakpoints` object that contains the site's configuration.
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 8]
```diff
@@ -1,21 +1,25 @@
...
 of the Bodiless Design System. If you are unfamiliar with Tailwind, you can
-learn more from one of
+learn more from:
 
 - [Tailwind on Github](https://github.com/tailwindcss/tailwindcss)
 - [Tailwind Docs](https://tailwindcss.com/docs/what-is-tailwind)
 
+Most of the site's styling can be defined with Tailwind and for the most part,
+BodilessJS will defer to Tailwind documentation. The following documentation
+either are specific Bodiless applications of Tailwind or important enough to
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:14]:
```suggestion
is either specific to Bodiless applications of Tailwind or important enough to
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 9]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
...
+build to see the changes.
+
+## Responsiveness with Tailwind
+
+Tailwind classes controls most of the responsiveness behavior of the site and
+classes can be prefixed with responsive size.
+
+### Breakpoints
+
+For more information, about setting the [responsive
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:67]:
```suggestion
For more information, see [responsive
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 10]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
...
+breakpoints](./Responsiveness#Breakpoints).
+
+## Setting maximum container width of a site
+
+If a site has maximum container width of a site should be, they can be set with
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:72]:
```suggestion
If a site has a maximum container width, they can be set with
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 11]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
...
+those classes instead of the tailwind classes.  
+
+The custom css files can be included imported either in gatsby-browser.js file,
+within the pages, or within the component that is using them.
+
+?> **Recommendation** is to keep the css file as close to possible to the
+component or page it is needed for.
+
+For example:
+* If its css used exclusively for the homepage, place it `/src/pages` and
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:99]:
```suggestion
* css used exclusively for the homepage should be placed in `/src/pages` and
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 12]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
...
+  include it in the `index.tsx`.
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:100]:
```suggestion
  included it in the `index.tsx`.
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 13]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
...
+* If its css used for a component, place it components folder and include it
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:101]:
```suggestion
* css used for a component should be placed in components folder and included
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 14]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
...
+  with the component.
+
+By doing the above, this custom css will only be loaded for pages or pages that
+use the component and help with performance. While BodilessJS runs with
+[gatsby purge css feature](https://www.gatsbyjs.org/packages/gatsby-plugin-purgecss)
+tool this is only processing on tailwind css and not on any custom css included.
+
+?> **Tip** As a site developer its always good practice to remove css that isn't
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:109]:
```suggestion
?> **Tip** As a site developer it is always good practice to remove css that isn't
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 15]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
...
+?> **Recommendation** is to keep the css file as close to possible to the
+component or page it is needed for.
+
+For example:
+* css used exclusively for the homepage should be placed in `/src/pages` and
+  included it in the `index.tsx`.
+* css used for a component should be placed in components folder and included
+  with the component.
+
+By doing the above, this custom css will only be loaded for pages or pages that
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:104]:
```suggestion
By doing the above, this custom css will only be loaded for pages that
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 16]
```diff
@@ -0,0 +1,83 @@
+# Responsiveness
+
+BodilessJS is in itself is not very opinionated about the responsiveness but there
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:3]:
```suggestion
BodilessJS in itself is not very opinionated about the responsiveness but there
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md [hunk 17]
```diff
@@ -48,3 +53,67 @@ Each time site builder makes a change in tailwind.config.js, she will need to
 rerun the build process. This is done in either `npm run start` or
 `npm run build` and will generate css that is automatically included for the
 site.
+
+!> **Important** for tailwind config changes to be updated, you must restart or
+build to see the changes.
+
+## Responsiveness with Tailwind
+
+Tailwind classes controls most of the responsiveness behavior of the site and
```
Comment (malachye) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/TailwindGuide.md:62]:
```suggestion
Tailwind classes control most of the responsiveness behavior of the site and
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+and
+[Starter Site](https://github.com/johnsonandjohnson/Bodiless-JS/tree/master/examples/starter).
+
+### Site Editing
+
+The BodilessJS site editing interface was designed for desktop
+scenario based on the assumption that site editors were working on desktop size
+screen for productivity.
+
+!> **Important** The BodilessJS editor and its forms are designed to work
```
Comment (VancheeZze) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:19]: Is `!>` a part of the markdown syntax?

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+
+| Small     | Medium    | Large      | Extra Large | 
+| --------- | --------- | ---------- | ----------- |
+| <= 640px  | 641-768px | 769-1024px | 1280px      |
+
+A site builder can change the breakpoints by modifying the site's
+`tailwind.config.js` by following the directions specified in
+[Tailwind Breakpoints](https://tailwindcss.com/docs/breakpoints/).
+
+?> **Tip** If a site developer does modify and/or add extra breakpoints, please
```
Comment (VancheeZze) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Responsiveness.md:36]: Same here: `?>`

**Pull request merged**